deker/schemas.py

```diff
@@ -314,44 +314,70 @@ class VArraySchema(SelfLoggerMixin, BaseArraysSchema):
 
     :param vgrid: an ordered sequence of positive integers; used for splitting ``VArray`` into ordinary ``Arrays``.
 
+      Each VArray dimension "size" shall be divided by the correspondent integer without remainders, thus an
+      ``Array's`` shape is created. If there is no need to split any dimension, its vgrid positional integer
+      shall be ``1``.
+
+    :param arrays_shape: an ordered sequence of positive integers; used for setting the shape of ordinary ``Arrays``
+      laying under a ``VArray``.
+
       Each integer in the sequence represents the total quantity of cells in the correspondent dimension
-      of each ordinary array. Each varray dimension "size" shall be divided by the correspondent integer
-      without remainders. If there is no need to split any dimension, its vgrid positional integer shall be ``1``.
+      of each ordinary ``Array``. Each ``VArray`` dimension "size" shall be divided by the correspondent integer
+      without remainders, thus a ``VArray's`` vgrid is created.
 
     :param fill_value: an optional value for filling in empty cells;
       If ``None`` - default value for each dtype will be used. Numpy ``nan`` can be used only for floating numpy dtypes.
     """
 
-    vgrid: Union[List[int], Tuple[int, ...]]
+    vgrid: Optional[Union[List[int], Tuple[int, ...]]] = None
+    arrays_shape: Optional[Union[List[int], Tuple[int, ...]]] = None
     fill_value: Union[Numeric, type(np.nan), None] = None  # type: ignore[valid-type]
     attributes: Union[List[AttributeSchema], Tuple[AttributeSchema, ...]] = tuple()
 
     def __attrs_post_init__(self) -> None:
         """Validate schema, convert vgrid to tuple and calculate arrays_shape."""
         super().__attrs_post_init__()
         __common_arrays_attributes_post_init__(self)
-        if (
-            not isinstance(self.vgrid, (list, tuple))
-            or not self.vgrid
-            or len(self.vgrid) != len(self.dimensions)
-            or not all(isinstance(g, int) for g in self.vgrid)
-            or any(g < 1 for g in self.vgrid)
-        ):
-            raise DekerValidationError(
-                f"Vgrid shall be a list or tuple of positive integers, with total elements quantity "
-                f"equal to dimensions quantity ({len(self.dimensions)}) "
-            )
-        for n, dim in enumerate(self.dimensions):
-            remainder = dim.size % self.vgrid[n]  # type: ignore[valid-type]
-            if remainder != 0:
+
+        splitters = {
+            attr: getattr(self, attr) for attr in ("vgrid", "arrays_shape") if getattr(self, attr)
+        }
+        if len(splitters) < 1:
+            raise DekerValidationError("Either `vgrid` or `arrays_shape` shall be passed")
+        if len(splitters) > 1:
+            raise DekerValidationError("Either `vgrid` or `arrays_shape` shall be passed, not both")
+        splitter, value = tuple(splitters.items())[0]
```

Comment:
Same logic as in adapters ? Can't we have it in one place ?

Reply:
[8d74073](https://github.com/openweathermap/deker/pull/8/commits/8d74073d4edf9fa75fc8afa880d5ff2e61af1cd7)